src/IssueViz.Security/HotspotsList2/ViewModels/HotspotsControlViewModel.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows.Data;
+using System.Windows.Input;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Security.SelectionService;
+using SonarLint.VisualStudio.IssueVisualization.Security.Store;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList2.ViewModels
+{
+    internal interface IHotspotsControlViewModel : IDisposable
+    {
+        ObservableCollection<IHotspotViewModel> Hotspots { get; }
+
+        IHotspotViewModel SelectedHotspot { get; }
+
+        ICommand NavigateCommand { get; }
+    }
+
+    internal sealed class HotspotsControlViewModel : IHotspotsControlViewModel, INotifyPropertyChanged
+    {
+        private readonly object Lock = new object();
+        private readonly INotifyCollectionChanged readonlyObservableHotspotsCollection;
+        private readonly IHotspotsSelectionService selectionService;
+
+        public ObservableCollection<IHotspotViewModel> Hotspots { get; } = new ObservableCollection<IHotspotViewModel>();
+
+        public IHotspotViewModel SelectedHotspot { get; set; }
+
+        public ICommand NavigateCommand { get; }
+
+        public HotspotsControlViewModel(IHotspotsStore hotspotsStore, ICommand navigateCommand, IHotspotsSelectionService selectionService)
+        {
+            BindingOperations.EnableCollectionSynchronization(Hotspots, Lock);
```

Comment:
To allow the hotspotsStore to be changed on a different thread.

Reply:
This is unusual enough to be worth calling out in comment.
The docs also say that it should be called on the UI thread, so it would be worth checking that with the ThreadHelper too.